// victory-native/lib/components/victory-polar-axis.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VictoryPolarAxis = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _reactNativeSvg = require("react-native-svg");
var _es = require("victory-polar-axis/es");
var _victoryLabel = require("./victory-label");
var _victoryContainer = require("./victory-container");
var _arc = require("./victory-primitives/arc");
var _lineSegment = require("./victory-primitives/line-segment");
var _wrapCoreComponent = require("../helpers/wrap-core-component");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const VictoryPolarAxis = exports.VictoryPolarAxis = (0, _wrapCoreComponent.wrapCoreComponent)({
  Component: _es.VictoryPolarAxis,
  defaultProps: {
    ..._es.VictoryPolarAxis.defaultProps,
    axisComponent: /*#__PURE__*/_react.default.createElement(_lineSegment.LineSegment, null),
    axisLabelComponent: /*#__PURE__*/_react.default.createElement(_victoryLabel.VictoryLabel, null),
    circularAxisComponent: /*#__PURE__*/_react.default.createElement(_arc.Arc, {
      type: "axis"
    }),
    circularGridComponent: /*#__PURE__*/_react.default.createElement(_arc.Arc, {
      type: "grid"
    }),
    tickLabelComponent: /*#__PURE__*/_react.default.createElement(_victoryLabel.VictoryLabel, null),
    tickComponent: /*#__PURE__*/_react.default.createElement(_lineSegment.LineSegment, null),
    gridComponent: /*#__PURE__*/_react.default.createElement(_lineSegment.LineSegment, null),
    containerComponent: /*#__PURE__*/_react.default.createElement(_victoryContainer.VictoryContainer, null),
    groupComponent: /*#__PURE__*/_react.default.createElement(_reactNativeSvg.G, null),
    width: _reactNative.Dimensions.get("window").width
  }
});
