// victory-native/lib/components/victory-voronoi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VictoryVoronoi = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _reactNativeSvg = require("react-native-svg");
var _es = require("victory-voronoi/es");
var _victoryLabel = require("./victory-label");
var _victoryContainer = require("./victory-container");
var _voronoi = require("./victory-primitives/voronoi");
var _wrapCoreComponent = require("../helpers/wrap-core-component");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const VictoryVoronoi = exports.VictoryVoronoi = (0, _wrapCoreComponent.wrapCoreComponent)({
  Component: _es.VictoryVoronoi,
  defaultProps: {
    ..._es.VictoryVoronoi.defaultProps,
    dataComponent: /*#__PURE__*/_react.default.createElement(_voronoi.Voronoi, null),
    labelComponent: /*#__PURE__*/_react.default.createElement(_victoryLabel.VictoryLabel, null),
    containerComponent: /*#__PURE__*/_react.default.createElement(_victoryContainer.VictoryContainer, null),
    groupComponent: /*#__PURE__*/_react.default.createElement(_reactNativeSvg.G, null),
    width: _reactNative.Dimensions.get("window").width
  }
});
