// victory-native/lib/components/victory-scatter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VictoryScatter = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _reactNativeSvg = require("react-native-svg");
var _es = require("victory-scatter/es");
var _victoryLabel = require("./victory-label");
var _victoryContainer = require("./victory-container");
var _point = require("./victory-primitives/point");
var _wrapCoreComponent = require("../helpers/wrap-core-component");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const VictoryScatter = exports.VictoryScatter = (0, _wrapCoreComponent.wrapCoreComponent)({
  Component: _es.VictoryScatter,
  defaultProps: {
    ..._es.VictoryScatter.defaultProps,
    dataComponent: /*#__PURE__*/_react.default.createElement(_point.Point, null),
    labelComponent: /*#__PURE__*/_react.default.createElement(_victoryLabel.VictoryLabel, null),
    containerComponent: /*#__PURE__*/_react.default.createElement(_victoryContainer.VictoryContainer, null),
    groupComponent: /*#__PURE__*/_react.default.createElement(_reactNativeSvg.G, null),
    width: _reactNative.Dimensions.get("window").width
  }
});
