// victory-native/lib/components/victory-errorbar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VictoryErrorBar = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _reactNativeSvg = require("react-native-svg");
var _es = require("victory-errorbar/es");
var _victoryContainer = require("./victory-container");
var _errorBar = require("./victory-primitives/error-bar");
var _wrapCoreComponent = require("../helpers/wrap-core-component");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const VictoryErrorBar = exports.VictoryErrorBar = (0, _wrapCoreComponent.wrapCoreComponent)({
  Component: _es.VictoryErrorBar,
  defaultProps: {
    ..._es.VictoryErrorBar.defaultProps,
    dataComponent: /*#__PURE__*/_react.default.createElement(_errorBar.ErrorBar, null),
    containerComponent: /*#__PURE__*/_react.default.createElement(_victoryContainer.VictoryContainer, null),
    groupComponent: /*#__PURE__*/_react.default.createElement(_reactNativeSvg.G, null),
    width: _reactNative.Dimensions.get("window").width
  }
});
