// victory-native/lib/components/victory-area.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VictoryArea = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _es = require("victory-area/es");
var _victoryLabel = require("./victory-label");
var _victoryContainer = require("./victory-container");
var _victoryClipContainer = require("./victory-clip-container");
var _area = require("./victory-primitives/area");
var _wrapCoreComponent = require("../helpers/wrap-core-component");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const VictoryArea = exports.VictoryArea = (0, _wrapCoreComponent.wrapCoreComponent)({
  Component: _es.VictoryArea,
  defaultProps: {
    ..._es.VictoryArea.defaultProps,
    dataComponent: /*#__PURE__*/_react.default.createElement(_area.Area, null),
    labelComponent: /*#__PURE__*/_react.default.createElement(_victoryLabel.VictoryLabel, null),
    containerComponent: /*#__PURE__*/_react.default.createElement(_victoryContainer.VictoryContainer, null),
    groupComponent: /*#__PURE__*/_react.default.createElement(_victoryClipContainer.VictoryClipContainer, null),
    width: _reactNative.Dimensions.get("window").width
  }
});
