// victory-native/lib/components/victory-histogram.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VictoryHistogram = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _reactNativeSvg = require("react-native-svg");
var _victoryLabel = require("./victory-label");
var _victoryContainer = require("./victory-container");
var _bar = require("./victory-primitives/bar");
var _es = require("victory-histogram/es");
var _wrapCoreComponent = require("../helpers/wrap-core-component");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const VictoryHistogram = exports.VictoryHistogram = (0, _wrapCoreComponent.wrapCoreComponent)({
  Component: _es.VictoryHistogram,
  defaultProps: {
    ..._es.VictoryHistogram.defaultProps,
    dataComponent: /*#__PURE__*/_react.default.createElement(_bar.Bar, null),
    labelComponent: /*#__PURE__*/_react.default.createElement(_victoryLabel.VictoryLabel, null),
    containerComponent: /*#__PURE__*/_react.default.createElement(_victoryContainer.VictoryContainer, null),
    groupComponent: /*#__PURE__*/_react.default.createElement(_reactNativeSvg.G, null),
    width: _reactNative.Dimensions.get("window").width
  }
});
