// victory-native/lib/components/victory-legend.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VictoryLegend = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNativeSvg = require("react-native-svg");
var _es = require("victory-legend/es");
var _reactNative = require("react-native");
var _victoryLabel = require("./victory-label");
var _victoryContainer = require("./victory-container");
var _point = require("./victory-primitives/point");
var _border = require("./victory-primitives/border");
var _wrapCoreComponent = require("../helpers/wrap-core-component");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const VictoryLegend = exports.VictoryLegend = (0, _wrapCoreComponent.wrapCoreComponent)({
  Component: _es.VictoryLegend,
  defaultProps: {
    ..._es.VictoryLegend.defaultProps,
    borderComponent: /*#__PURE__*/_react.default.createElement(_border.Border, null),
    containerComponent: /*#__PURE__*/_react.default.createElement(_victoryContainer.VictoryContainer, null),
    dataComponent: /*#__PURE__*/_react.default.createElement(_point.Point, null),
    groupComponent: /*#__PURE__*/_react.default.createElement(_reactNativeSvg.G, null),
    labelComponent: /*#__PURE__*/_react.default.createElement(_victoryLabel.VictoryLabel, null),
    titleComponent: /*#__PURE__*/_react.default.createElement(_victoryLabel.VictoryLabel, null),
    width: _reactNative.Dimensions.get("window").width
  }
});
