// victory-native/lib/components/victory-pie.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VictoryPie = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _reactNativeSvg = require("react-native-svg");
var _es = require("victory-pie/es");
var _victoryLabel = require("./victory-label");
var _victoryContainer = require("./victory-container");
var _slice = require("./victory-primitives/slice");
var _wrapCoreComponent = require("../helpers/wrap-core-component");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const VictoryPie = exports.VictoryPie = (0, _wrapCoreComponent.wrapCoreComponent)({
  Component: _es.VictoryPie,
  defaultProps: {
    ..._es.VictoryPie.defaultProps,
    dataComponent: /*#__PURE__*/_react.default.createElement(_slice.Slice, null),
    labelComponent: /*#__PURE__*/_react.default.createElement(_victoryLabel.VictoryLabel, null),
    containerComponent: /*#__PURE__*/_react.default.createElement(_victoryContainer.VictoryContainer, null),
    groupComponent: /*#__PURE__*/_react.default.createElement(_reactNativeSvg.G, null),
    height: _reactNative.Dimensions.get("window").width,
    width: _reactNative.Dimensions.get("window").width
  }
});
