// victory-native/lib/components/victory-selection-container.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.selectionContainerMixin = exports.VictorySelectionContainer = void 0;
var _flow2 = _interopRequireDefault(require("lodash/flow"));
var _react = _interopRequireDefault(require("react"));
var _reactNativeSvg = require("react-native-svg");
var _victorySelectionContainer = require("victory-selection-container");
var _victoryContainer = require("./victory-container");
var _nativeHelpers = _interopRequireDefault(require("../helpers/native-helpers"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
// ensure the selection component get native styles
const DefaultSelectionComponent = _ref => {
  let {
    style = {},
    ...otherProps
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_reactNativeSvg.Rect, _extends({}, otherProps, _nativeHelpers.default.getStyle(style)));
};
function nativeSelectionMixin(Base) {
  // @ts-expect-error "TS2545: A mixin class must have a constructor with a single rest parameter of type 'any[]'."
  return class VictoryNativeSelectionContainer extends Base {
    // eslint-disable-line max-len
    // assign native specific defaultProps over web `VictorySelectionContainer` defaultProps
    static defaultProps = {
      ..._victorySelectionContainer.VictorySelectionContainer.defaultProps,
      standalone: true,
      selectionComponent: /*#__PURE__*/_react.default.createElement(DefaultSelectionComponent, null)
    };

    // overrides all web events with native specific events
    static defaultEvents(props) {
      return [{
        target: "parent",
        eventHandlers: {
          onTouchStart: (evt, targetProps) => {
            if (props.disable) {
              return {};
            }
            _victorySelectionContainer.SelectionHelpers.onMouseMove.cancel();
            return _victorySelectionContainer.SelectionHelpers.onMouseDown(evt, targetProps);
          },
          onTouchMove: (evt, targetProps) => {
            return props.disable ? {} : _victorySelectionContainer.SelectionHelpers.onMouseMove(evt, targetProps);
          },
          onTouchEnd: (evt, targetProps) => {
            if (props.disable) {
              return {};
            }
            _victorySelectionContainer.SelectionHelpers.onMouseMove.cancel();
            return _victorySelectionContainer.SelectionHelpers.onMouseUp(evt, targetProps);
          }
        }
      }];
    }
  };
}
const combinedMixin = (0, _flow2.default)(_victorySelectionContainer.selectionContainerMixin, nativeSelectionMixin);
const selectionContainerMixin = base => combinedMixin(base);
exports.selectionContainerMixin = selectionContainerMixin;
const VictorySelectionContainer = exports.VictorySelectionContainer = selectionContainerMixin(_victoryContainer.VictoryContainer);
