// victory-native/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  VictoryAnimation: true,
  VictoryTheme: true,
  VictoryTransition: true,
  addEvents: true,
  Collection: true,
  Data: true,
  DefaultTransitions: true,
  Domain: true,
  Events: true,
  Helpers: true,
  Log: true,
  Scale: true,
  Style: true,
  TextSize: true,
  Transitions: true,
  Selection: true,
  LabelHelpers: true,
  Axis: true,
  Wrapper: true,
  VictorySharedEvents: true,
  NativeHelpers: true,
  NativeZoomHelpers: true,
  createContainer: true
};
Object.defineProperty(exports, "Axis", {
  enumerable: true,
  get: function () {
    return _victoryCore.Axis;
  }
});
Object.defineProperty(exports, "Collection", {
  enumerable: true,
  get: function () {
    return _victoryCore.Collection;
  }
});
Object.defineProperty(exports, "Data", {
  enumerable: true,
  get: function () {
    return _victoryCore.Data;
  }
});
Object.defineProperty(exports, "DefaultTransitions", {
  enumerable: true,
  get: function () {
    return _victoryCore.DefaultTransitions;
  }
});
Object.defineProperty(exports, "Domain", {
  enumerable: true,
  get: function () {
    return _victoryCore.Domain;
  }
});
Object.defineProperty(exports, "Events", {
  enumerable: true,
  get: function () {
    return _victoryCore.Events;
  }
});
Object.defineProperty(exports, "Helpers", {
  enumerable: true,
  get: function () {
    return _victoryCore.Helpers;
  }
});
Object.defineProperty(exports, "LabelHelpers", {
  enumerable: true,
  get: function () {
    return _victoryCore.LabelHelpers;
  }
});
Object.defineProperty(exports, "Log", {
  enumerable: true,
  get: function () {
    return _victoryCore.Log;
  }
});
Object.defineProperty(exports, "NativeHelpers", {
  enumerable: true,
  get: function () {
    return _nativeHelpers.default;
  }
});
Object.defineProperty(exports, "NativeZoomHelpers", {
  enumerable: true,
  get: function () {
    return _nativeZoomHelpers.default;
  }
});
Object.defineProperty(exports, "Scale", {
  enumerable: true,
  get: function () {
    return _victoryCore.Scale;
  }
});
Object.defineProperty(exports, "Selection", {
  enumerable: true,
  get: function () {
    return _victoryCore.Selection;
  }
});
Object.defineProperty(exports, "Style", {
  enumerable: true,
  get: function () {
    return _victoryCore.Style;
  }
});
Object.defineProperty(exports, "TextSize", {
  enumerable: true,
  get: function () {
    return _victoryCore.TextSize;
  }
});
Object.defineProperty(exports, "Transitions", {
  enumerable: true,
  get: function () {
    return _victoryCore.Transitions;
  }
});
Object.defineProperty(exports, "VictoryAnimation", {
  enumerable: true,
  get: function () {
    return _victoryCore.VictoryAnimation;
  }
});
Object.defineProperty(exports, "VictorySharedEvents", {
  enumerable: true,
  get: function () {
    return _victorySharedEvents.VictorySharedEvents;
  }
});
Object.defineProperty(exports, "VictoryTheme", {
  enumerable: true,
  get: function () {
    return _victoryCore.VictoryTheme;
  }
});
Object.defineProperty(exports, "VictoryTransition", {
  enumerable: true,
  get: function () {
    return _victoryCore.VictoryTransition;
  }
});
Object.defineProperty(exports, "Wrapper", {
  enumerable: true,
  get: function () {
    return _victoryCore.Wrapper;
  }
});
Object.defineProperty(exports, "addEvents", {
  enumerable: true,
  get: function () {
    return _victoryCore.addEvents;
  }
});
Object.defineProperty(exports, "createContainer", {
  enumerable: true,
  get: function () {
    return _createContainer.createContainer;
  }
});
var _victoryCore = require("victory-core");
var _victorySharedEvents = require("victory-shared-events");
var _circle = require("./components/victory-primitives/circle");
Object.keys(_circle).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _circle[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _circle[key];
    }
  });
});
var _line = require("./components/victory-primitives/line");
Object.keys(_line).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _line[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _line[key];
    }
  });
});
var _path = require("./components/victory-primitives/path");
Object.keys(_path).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _path[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _path[key];
    }
  });
});
var _rect = require("./components/victory-primitives/rect");
Object.keys(_rect).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _rect[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _rect[key];
    }
  });
});
var _text = require("./components/victory-primitives/text");
Object.keys(_text).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _text[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _text[key];
    }
  });
});
var _tspan = require("./components/victory-primitives/tspan");
Object.keys(_tspan).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _tspan[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _tspan[key];
    }
  });
});
var _arc = require("./components/victory-primitives/arc");
Object.keys(_arc).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _arc[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _arc[key];
    }
  });
});
var _area = require("./components/victory-primitives/area");
Object.keys(_area).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _area[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _area[key];
    }
  });
});
var _background = require("./components/victory-primitives/background");
Object.keys(_background).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _background[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _background[key];
    }
  });
});
var _bar = require("./components/victory-primitives/bar");
Object.keys(_bar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _bar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _bar[key];
    }
  });
});
var _border = require("./components/victory-primitives/border");
Object.keys(_border).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _border[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _border[key];
    }
  });
});
var _candle = require("./components/victory-primitives/candle");
Object.keys(_candle).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _candle[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _candle[key];
    }
  });
});
var _clipPath = require("./components/victory-primitives/clip-path");
Object.keys(_clipPath).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _clipPath[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _clipPath[key];
    }
  });
});
var _curve = require("./components/victory-primitives/curve");
Object.keys(_curve).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _curve[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _curve[key];
    }
  });
});
var _errorBar = require("./components/victory-primitives/error-bar");
Object.keys(_errorBar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _errorBar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _errorBar[key];
    }
  });
});
var _lineSegment = require("./components/victory-primitives/line-segment");
Object.keys(_lineSegment).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _lineSegment[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _lineSegment[key];
    }
  });
});
var _point = require("./components/victory-primitives/point");
Object.keys(_point).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _point[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _point[key];
    }
  });
});
var _slice = require("./components/victory-primitives/slice");
Object.keys(_slice).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _slice[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _slice[key];
    }
  });
});
var _voronoi = require("./components/victory-primitives/voronoi");
Object.keys(_voronoi).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _voronoi[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _voronoi[key];
    }
  });
});
var _flyout = require("./components/victory-primitives/flyout");
Object.keys(_flyout).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _flyout[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _flyout[key];
    }
  });
});
var _whisker = require("./components/victory-primitives/whisker");
Object.keys(_whisker).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _whisker[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _whisker[key];
    }
  });
});
var _victoryTooltip = require("./components/victory-tooltip");
Object.keys(_victoryTooltip).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryTooltip[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryTooltip[key];
    }
  });
});
var _victoryPortal = require("./components/victory-portal/victory-portal");
Object.keys(_victoryPortal).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryPortal[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryPortal[key];
    }
  });
});
var _portal = require("./components/victory-portal/portal");
Object.keys(_portal).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _portal[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _portal[key];
    }
  });
});
var _victoryArea = require("./components/victory-area");
Object.keys(_victoryArea).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryArea[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryArea[key];
    }
  });
});
var _victoryAxis = require("./components/victory-axis");
Object.keys(_victoryAxis).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryAxis[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryAxis[key];
    }
  });
});
var _victoryPolarAxis = require("./components/victory-polar-axis");
Object.keys(_victoryPolarAxis).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryPolarAxis[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryPolarAxis[key];
    }
  });
});
var _victoryBar = require("./components/victory-bar");
Object.keys(_victoryBar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryBar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryBar[key];
    }
  });
});
var _victoryBrushLine = require("./components/victory-brush-line");
Object.keys(_victoryBrushLine).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryBrushLine[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryBrushLine[key];
    }
  });
});
var _victoryBoxplot = require("./components/victory-boxplot");
Object.keys(_victoryBoxplot).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryBoxplot[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryBoxplot[key];
    }
  });
});
var _victoryGroup = require("./components/victory-group");
Object.keys(_victoryGroup).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryGroup[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryGroup[key];
    }
  });
});
var _victoryLine = require("./components/victory-line");
Object.keys(_victoryLine).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryLine[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryLine[key];
    }
  });
});
var _victoryHistogram = require("./components/victory-histogram");
Object.keys(_victoryHistogram).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryHistogram[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryHistogram[key];
    }
  });
});
var _victoryScatter = require("./components/victory-scatter");
Object.keys(_victoryScatter).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryScatter[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryScatter[key];
    }
  });
});
var _victoryStack = require("./components/victory-stack");
Object.keys(_victoryStack).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryStack[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryStack[key];
    }
  });
});
var _victoryChart = require("./components/victory-chart");
Object.keys(_victoryChart).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryChart[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryChart[key];
    }
  });
});
var _victoryErrorbar = require("./components/victory-errorbar");
Object.keys(_victoryErrorbar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryErrorbar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryErrorbar[key];
    }
  });
});
var _victoryCandlestick = require("./components/victory-candlestick");
Object.keys(_victoryCandlestick).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryCandlestick[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryCandlestick[key];
    }
  });
});
var _victoryVoronoi = require("./components/victory-voronoi");
Object.keys(_victoryVoronoi).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryVoronoi[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryVoronoi[key];
    }
  });
});
var _victoryPie = require("./components/victory-pie");
Object.keys(_victoryPie).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryPie[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryPie[key];
    }
  });
});
var _victoryContainer = require("./components/victory-container");
Object.keys(_victoryContainer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryContainer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryContainer[key];
    }
  });
});
var _victoryClipContainer = require("./components/victory-clip-container");
Object.keys(_victoryClipContainer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryClipContainer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryClipContainer[key];
    }
  });
});
var _victoryLabel = require("./components/victory-label");
Object.keys(_victoryLabel).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryLabel[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryLabel[key];
    }
  });
});
var _victoryLegend = require("./components/victory-legend");
Object.keys(_victoryLegend).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryLegend[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryLegend[key];
    }
  });
});
var _victoryZoomContainer = require("./components/victory-zoom-container");
Object.keys(_victoryZoomContainer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryZoomContainer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryZoomContainer[key];
    }
  });
});
var _victoryVoronoiContainer = require("./components/victory-voronoi-container");
Object.keys(_victoryVoronoiContainer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryVoronoiContainer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryVoronoiContainer[key];
    }
  });
});
var _victorySelectionContainer = require("./components/victory-selection-container");
Object.keys(_victorySelectionContainer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victorySelectionContainer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victorySelectionContainer[key];
    }
  });
});
var _victoryCursorContainer = require("./components/victory-cursor-container");
Object.keys(_victoryCursorContainer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryCursorContainer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryCursorContainer[key];
    }
  });
});
var _victoryBrushContainer = require("./components/victory-brush-container");
Object.keys(_victoryBrushContainer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _victoryBrushContainer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _victoryBrushContainer[key];
    }
  });
});
var _nativeHelpers = _interopRequireDefault(require("./helpers/native-helpers"));
var _nativeZoomHelpers = _interopRequireDefault(require("./helpers/native-zoom-helpers"));
var _createContainer = require("./helpers/create-container");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
